### ISSUE TITLE
Namespaces are a valid alternative to prefixing for constants

### ISSUE THREAD

**Issue by grappler** [WordPress-Coding-Standards/WordPress-Coding-Standards#1056]
I have the following code in a plugin:

`define( __NAMESPACE__ . '\PLUGIN_FILE', __FILE__ );`

I am getting a warning for this `WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedConstantFound`

Namespaces are a valid alternative to prefixing.

Related to #764

**Comment by jrfnl** [comment]
@grappler Good point. I've got a PR ready for this, but will wait to pull this until 0.13.0 has been released.

**Comment by GaryJones** [comment]
Related, is this false positive:

```php
add_action( 'after_setup_theme', __NAMESPACE__ . '\\load_textdomain', 5 );
/**
 * Load theme text domain.
 *
 * @since 1.0.0
 */
function load_textdomain() : void {
	\load_child_theme_textdomain( 'my-textdomain', get_stylesheet_directory() . '/languages' );
}
```

Gives:

```sh
94 | ERROR   | Functions declared by a theme/plugin should start with the theme/plugin prefix. Found: "load_textdomain". (WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedFunctionFound)
```

**Comment by jrfnl** [comment]
@GaryJones I presume your code sample is missing a `namespace MyNamespace;` at the top ? 

If not, this is not a false positive as without a namespace at the top of the file (or a scoped namespace) `__NAMESPACE__` resolves to `''`, i.e. global namespace.

**Comment by GaryJones** [comment]
> I presume your code sample is missing a namespace MyNamespace; at the top ?

No.... it seems my file didn't have the namespace declared...

Sorry 😊

**Comment by jrfnl** [comment]
@GaryJones Happens to the best of us ;-)
